src/app/config/schema.ts

```diff
@@ -436,6 +437,39 @@ export const prodOnlyVarsSchema: Schema<IProdOnlyVarsSchema> = {
     env: 'DB_HOST',
     sensitive: true,
   },
+  // Rename these env vars to without the _sg suffix when SES migration is over (opengovsg/formsg-private#130)
+  port_sg: {
+    doc: 'SMTP port number',
+    format: 'port',
+    default: null,
+    env: 'SES_PORT_SG',
+  },
+  host_sg: {
+    doc: 'SMTP hostname',
+    format: String,
+    default: null,
+    env: 'SES_HOST_SG',
+  },
+  user_sg: {
+    doc: 'SMTP username',
+    format: String,
+    default: null,
+    env: 'SES_USER_SG',
+  },
+  pass_sg: {
+    doc: 'SMTP password',
+    format: String,
+    default: null,
+    env: 'SES_PASS_SG',
+    sensitive: true,
+  },
+  // Remove this when SES migration is over (opengovsg/formsg-private#130)
+  nodemailer_client_threshold_sg: {
+    doc: 'Percentage threshold to send mail from SG nodemailer client',
+    format: 'int',
+    default: 0,
+    env: 'NODEMAILER_THRESHOLD_SG',
+  },
```

Comment:
Apologies, I'm making a change to the requirement 😅😰.

This threshold is like the thresholds we had for the react migration. In react setting the actual threshold value manually made sense, because we wanted high-control deliberate steps, and no change over the weekends.

For SES, the warm up would be better served with a gradual and continuous increase over the fixed warm-up period of 6 weeks (see doc [here](https://docs.aws.amazon.com/ses/latest/dg/dedicated-ip-warming.html)). 

Therefore, I'd recommend new requirement/behaviour as follows:
1. env variable represents whether the split is active at all (so we can turn it off if needed)
2. env var value is a date of start of warm up as `YYYY-MM-DDTHH:MM:SS`
3. hardcode 6 WEEKS as a constant in code
4 derive threshold from a function (which basically embeds the rollout strategy)

Example linear growth (pseudo code);
```javascript
const START_TS = (new Date(env.SES_US_SG_START_DATE)).getTime()
const WARM_UP_DURATION = 6 * 7 * 24 * 60 * 60 * 1000; // 6 weeks

// return threshold on [0, 1]
function getThreshold() {
    const now = Date.now()
    const elapsed = Math.max(0, Math.min(now - START_TS, WARM_UP_DURATION))
    const linear = elapsed / WARM_UP_DURATION

    return linear
}
```

Or quadratic growth (slower start, faster end):
```javascript
const START_TS = (new Date(env.SES_US_SG_START_DATE)).getTime()
const WARM_UP_DURATION = 6 * 7 * 24 * 60 * 60 * 1000; // 6 weeks

// return threshold on [0, 1]
function getThreshold() {
    const now = Date.now()
    const elapsed = Math.max(0, Math.min(now - START_TS, WARM_UP_DURATION))
    const linear = elapsed / WARM_UP_DURATION

    return linear * linear
}
```

Reply:
The system above can then be plugged in to the check that's already done in the PR:
```javascript
if (Math.random() < getThreshold()) {
    // SES SG
}
else {
    // SES US
}
```

Reply:
ok! this approach does make more sense, will make the change